atom/browser/api/lib/app.coffee

```diff
@@ -25,6 +25,12 @@ if process.platform is 'darwin'
     show: bindings.dockShow
     setMenu: bindings.dockSetMenu
 
+app._setAppPath = (path) ->
+  this.path = path
```

Comment:
You can put `path` outside `app` object to prevent accidental modifications from users:

```js
appPath = null
app._setAppPath = (path) ->
  appPath = path
```